File: binarySearchRecursion.cpp

#include<bits/stdc++.h>
using namespace std;

int binarySearch(int arr[],int l,int r,int x){
    if(l <= r){
        int m = (l + (r-1))/2;
        if (arr[m] == x)
        {
            return m+1;
        }
        if (arr[m] > x)
        {
            return binarySearch(arr, l, m - 1, x);
        }
        return binarySearch(arr, m + 1, r , x);
        
    }
    return -1;
}

//driver code 
int main(){
    //intialize
    int a[] = {4,7,9,12,56};
    int x = 12;
    int n = sizeof(a)/sizeof(a[0]);

    int result = binarySearch(a,0,n-1,x);
    (result == -1)
        ? cout << "NOT PRESENT IN THE ARRAY "
        : cout << "PRESNT AT POSITION " << result;
    return 0;
}

File: binarySearchRecursion.java
public class binarySearchRecursion {
    static int binSearch(int arr[], int l, int r, int x)
    {
        while(l<=r){
            int mid = (l+r)/2;
            if(arr[mid] == x)
                return mid;
            else if(arr[mid] > x)
                return binSearch(arr,l,mid-1,x);
            else if(arr[mid] < x)
                return binSearch(arr,mid+1,r,x);

        }
        return -1;
    }
    public static void main(String[] args) {
        int arr[] = {12,23,34,45,56,67};
        int x = 67;
        int n = arr.length;
        int result = binSearch(arr,0,(n-1),x);
        if(result != -1)
            System.out.println("the number "+x+" is found at position "+(result+1));
        else
            System.out.println("result is not found");
    }
}
